test/unit/test.discovery.v1.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+var assert = require('assert');
+var Discovery = require('../../discovery/v1');
+var fs = require('fs');
+var path = require('path');
+
+var nock   = require('nock');
+
+describe('discovery-v1', function() {
+
+  var noop = function() {};
+
+  // Test params
+  var service = {
+    username: 'batman',
+    password: 'bruce-wayne',
+    url: 'http://ibm.com:80',
+    version: 'v1-experimental',
+    version_date: '2016-11-07'
+  };
+
+  var service1 = {
+    password: 'bruce-wayne',
+    url: 'http://ibm.com:80',
+    version: 'v1',
+    username: 'batman'
+  };
+
+  var paths = {
+    environments : '/v1/environments',
+    environmentinfo : '/v1/environments/env-guid',
+    collections: '/v1/environments/env-guid/collections',
+    collectioninfo: '/v1/environments/env-guid/collections/col-guid',
+    configurations: '/v1/environments/env-guid/configurations',
+    configurationinfo: '/v1/environments/env-guid/configurations/config-guid',
+    delete_collection: '/v1/environments/env-guid/collections/col-guid',
+    add_document: '/v1/environments/env-guid/collections/col-guid/documents',
+    delete_document: '/v1/environments/env-guid/collections/col-guid/documents/document-guid',
+    query: '/v1/environments/env-guid/collections/col-guid/query'
+  };
+
+  before(function() {
+    nock.disableNetConnect();
+    nock(service.url)
+      .persist()
+      .post(paths.environments + '?version=' + service.version_date)
+      .reply(200, {'environment_id': 'yes'})
+      .get(paths.environmentinfo + '?version=' + service.version_date)
+      .reply(200, {'environment_id': 'info'})
+      .get(paths.collections + '?version=' + service.version_date)
+      .reply(200, {'collection_id': 'yes'})
+      .get(paths.collectioninfo + '?version=' + service.version_date)
+      .reply(200, {'collection_id': 'info'})
+      .get(paths.query + '?version=' + service.version_date)
+      .reply(200, {'query': 'yes'})
+      .post(paths.collections + '?version=' + service.version_date)
+      .reply(200, {'collection_id': 'yes'})
+      .delete(paths.delete_collection + '?version=' + service.version_date)
+      .reply(200, {'config': 'yes'})
+      .post(paths.add_document + '?version=' + service.version_date)
+      .reply(200, {'add_doc': 'yes'})
+      .delete(paths.delete_document + '?version=' + service.version_date)
+      .reply(200, {'delete_doc': 'yes'})
+      .get(paths.configurations + '?version=' + service.version_date)
+      .reply(200, {'configs': 'yes'});
```

Comment:
FYI: these don't hurt, but nock can't check things in time for for sync tests because the test passes before nock gets called, and any error that nock might throw gets swallowed by mocha because it occurs outside of a test :/